ivy.el

```diff
@@ -1496,55 +1496,58 @@ See variable `ivy-recursive-restore' for further information."
-            ((null action)
-             current)
-            ((eq action #'identity)
-             (prog1 x
-               (ivy-recursive-restore)))
-            (t
-             (select-window (ivy--get-window ivy-last))
-             (set-buffer (ivy-state-buffer ivy-last))
-             (prog1 (unwind-protect
-                         (if ivy-marked-candidates
-                             (let* ((prefix-len (length ivy-mark-prefix))
-                                    (marked-candidates
-                                     (mapcar
-                                      (lambda (s)
-                                        (let ((cand (substring s prefix-len)))
-                                          (if ivy--directory
-                                              (expand-file-name cand ivy--directory)
-                                            cand)))
-                                      ivy-marked-candidates))
-                                    (multi-action (ivy--get-multi-action ivy-last)))
-                               (if multi-action
-                                   (funcall multi-action marked-candidates)
-                                 (dolist (c marked-candidates)
-                                   (funcall action c))))
-                           (funcall action x))
-                      (ivy-recursive-restore))
-               (unless (or (eq ivy-exit 'done)
-                           (minibuffer-window-active-p (selected-window))
-                           (null (active-minibuffer-window)))
-                 (select-window (active-minibuffer-window))))))))
+           ((null action)
+            current)
+           ((eq action #'identity)
+            (prog1 x
+              (ivy-recursive-restore)))
+           (t
+            (select-window (ivy--get-window ivy-last))
+            (set-buffer (ivy-state-buffer ivy-last))
+            (prog1 (unwind-protect
+                       (if ivy-marked-candidates
+                           (let* ((prefix-len (length ivy-mark-prefix))
+                                  (marked-candidates
+                                   (mapcar
+                                    (lambda (s)
+                                      (let ((cand (substring s prefix-len)))
+                                        (if ivy--directory
+                                            (expand-file-name cand ivy--directory)
+                                          cand)))
+                                    ivy-marked-candidates))
+                                  (multi-action (ivy--get-multi-action ivy-last)))
+                             (if multi-action
+                                 (let ((default-directory (ivy-state-directory ivy-last)))
+                                   (funcall multi-action marked-candidates))
+                               (dolist (c marked-candidates)
+                                 (let ((default-directory (ivy-state-directory ivy-last)))
```

Comment:
Why not bind this only once outside of the `dolist`, or even outside of the surrounding `(if multi-action ...)`?

Reply:
Because the actions might change the default directory inside the loop.

Reply:
Ah right, sorry.